source/util/source/ftp_server.c

```
@@ -0,0 +1,2370 @@
+
+  // Create file list.
+  while ((dp = readdir(dfd)) != NULL) {
+    if (strcmp(dp->d_name, ".") == 0 || strcmp(dp->d_name, "..") == 0)
+      continue;
+
+    char *file_name = (char *)malloc(strlen(dp->d_name) + 3);
+    if (file_name == NULL)
+      goto out_of_memory;
+
+    sprintf(file_name, "%s" CRLF, dp->d_name);
+    flist = (char **)realloc(flist, sizeof(*flist) * ++flist_size);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)